plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogAccess.java

```diff
@@ -196,50 +197,47 @@ public void invalidateCaches(String tableLocation)
 
     public List<AddFileEntry> getActiveFiles(TableSnapshot tableSnapshot, ConnectorSession session)
     {
-        String tableLocation = tableSnapshot.getTableLocation().toString();
-        DeltaLakeDataFileCacheEntry cachedTable = activeDataFileCache.getIfPresent(tableLocation);
-
-        if (cachedTable == null || cachedTable.getVersion() > tableSnapshot.getVersion()) {
+        try {
```

Comment:
Off-topic: I was trying to understand where the method `getActiveFiles`  is being used and noticed that the `tableSnapshot` parameter is always produced by a previous call to `TransactionLogAccess` as well.

`transactionLogAccess.getMetadataEntry` depends also on  a `tableSnapshot` previously obtained from `TransactionLogAccess` as well.

It would be worth considering exposing a method which retrieves both the active metadata & files:

```
   // pseudocode
   MetadataAndAddFileEntryList getActiveMetadataAndAddFileEntryList(ConnectorSession session)
```

Reply:
I think it makes sense but maybe not in scope of this PR. Please create a separate ticket for that,

Reply:
https://github.com/trinodb/trino/issues/11590